fenix/app/src/test/java/org/mozilla/fenix/search/awesomebar/AwesomeBarViewTest.kt

```diff
@@ -180,16 +267,45 @@ class AwesomeBarViewTest {
                     every { resultsUrl } returns url
                 },
             ),
+            showSponsoredSuggestions = true,
         )
 
         val result = awesomeBarView.getProvidersToAdd(state)
 
         val historyProvider = result.firstOrNull { it is CombinedHistorySuggestionProvider }
         assertNotNull(historyProvider)
-        assertEquals(url, (historyProvider as CombinedHistorySuggestionProvider).resultsUriFilter)
+        assertNotNull((historyProvider as CombinedHistorySuggestionProvider).resultsUriFilter)
         assertEquals(AwesomeBarView.METADATA_SUGGESTION_LIMIT, historyProvider.getMaxNumberOfSuggestions())
     }
 
+    @Test
+    fun `GIVEN a search that should show filtered history WHEN history metadata is enabled and sponsored suggestions are disabled THEN return a history metadata provider with an engine filter`() {
+        val settings: Settings = mockk(relaxed = true) {
+            every { historyMetadataUIFeature } returns true
+        }
+        val url = Uri.parse("test.com")
+        every { activity.settings() } returns settings
+        val state = getSearchProviderState(
+            showAllHistorySuggestions = false,
+            searchEngineSource = SearchEngineSource.Shortcut(
+                mockk(relaxed = true) {
+                    every { resultsUrl } returns url
+                },
+            ),
+            showSponsoredSuggestions = false,
+        )
+
+        val result = awesomeBarView.getProvidersToAdd(state)
+
+        val historyProvider = result.firstOrNull { it is CombinedHistorySuggestionProvider }
+        assertNotNull(historyProvider)
+        assertNotNull((historyProvider as CombinedHistorySuggestionProvider).resultsUriFilter)
```

Comment:
I'm wondering if there's an easy way to test which filter has been applied just so that we can verify that we're not missing any existing functionality.

Reply:
I wonder if `::` gives us a stable lambda, so that we can check if it’s equal here. I‘ll check back later this evening, or tomorrow 😊